From a359a9a6edffa6a2cd3bf35ba15954ade3c8aa30 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Wed, 25 Dec 2019 09:36:06 +0100
Subject: [PATCH] enable discard option for dom0 filesystems by default

This may have performance impact on some older SSD, but on the other
hand, without this option it's pretty easy to fill the whole LVM thin
pool even if there is plenty free space in dom0.

Fixes QubesOS/qubes-issues#3226
---
 pyanaconda/modules/storage/devicetree/fsset.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pyanaconda/modules/storage/devicetree/fsset.py b/pyanaconda/modules/storage/devicetree/fsset.py
index af97dc3bba..fcd889e93b 100644
--- a/pyanaconda/modules/storage/devicetree/fsset.py
+++ b/pyanaconda/modules/storage/devicetree/fsset.py
@@ -774,6 +774,8 @@ class FSSet(object):
                     break
             if device.encrypted:
                 options += ",x-systemd.device-timeout=0"
+            if fstype in ('ext4', 'btrfs', 'xfs', 'vfat'):
+                options += ",discard"
             devspec = device.fstab_spec
             dump = device.format.dump
             if device.format.check and mountpoint == "/":
-- 
2.45.2

